webpack/InsightsCloudSync/Components/InsightsSyncSwitcher/InsightsSyncSwitcher.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Switch, FieldLevelHelp } from 'patternfly-react';
+import { translate as __ } from 'foremanReact/common/I18n';
+import './insightsSyncSwitcher.scss';
+
+class InsightsSyncSwitcher extends Component {
+  constructor(props) {
+    super(props);
+
+    // This binding is necessary to make `this` work in the callback
+    this.handleToggle = this.handleToggle.bind(this);
+  }
+
+  handleToggle() {
+    const { insightsSyncEnabled, setInsightsSyncEnabled } = this.props;
+
+    const toggledInsightsSyncEnabled = !insightsSyncEnabled;
+    setInsightsSyncEnabled(toggledInsightsSyncEnabled);
+  }
```

Comment:
arrow function should do the binding work for you:

```suggestion
handleToggle = () => {
    const { insightsSyncEnabled, setInsightsSyncEnabled } = this.props;

    const toggledInsightsSyncEnabled = !insightsSyncEnabled;
    setInsightsSyncEnabled(toggledInsightsSyncEnabled);
  }
```